packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1268,13 +1268,26 @@ export class InjectedScript {
 
       // Each matcher should get a "received" that matches it, in order.
       const matchers = options.expectedText.map(e => new ExpectedTextMatcher(e));
-      let mIndex = 0, rIndex = 0;
-      while (mIndex < matchers.length && rIndex < received.length) {
-        if (matchers[mIndex].matches(received[rIndex]))
-          ++mIndex;
-        ++rIndex;
+      if (options.ignoreOrder) {
```

Comment:
I'd just `sort()` if the order does not matter, and use the same code as before.